### PR TITLE
Move to Python 3.10 & improve CI workflow

### PR DESCRIPTION
**Pull request by Pwuts**
### Background
CI workflow was updated but now runs twice. Also runs with Python 3.8 which is inconsistent with README (#2342). Python version in other places is also inconsistent.

`pull_request_target` is both risky and unnecessary: CI will already run on `push` in the fork.

### Changes
* Split `build` workflow into `lint` and `test` workflows
* Run all checks even if some fail
* Fix double PR runs
* Adopted #2343
* Adopted #2107

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,26 +2,24 @@ name: Python CI
...
-    branches:
-      - master
+    branches: [master]
   pull_request:
-    branches:
-      - '**'
-  pull_request_target:
-    branches:
-      - '**'
+    branches: [master]
```
Comment (ntindle) [inline .github/workflows/ci.yml:7]: Add stable here

Comment (Pwuts) [reply .github/workflows/ci.yml:7]: No need with branch protection rules

**Comment by aperullo** [comment]
Related: https://github.com/Significant-Gravitas/Auto-GPT/pull/2267

**Comment by Androbin** [comment]
Don't forget to update `.github/workflows/benchmark.yml` and `.devcontainer/Dockerfile`, see #2107

**Comment by ntindle** [comment]
PR for #2424 pt 1: "shirt sizing" is ready pending this pull request being merged into master on `Pwuts/Auto-GPT/tree/ci/shirt-sizing`

**Review by BillSchumacher — APPROVED** [review]

**Pull request merged**